src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+        switch (random.nextInt(5)) {

```

Comment:
Consider defining the number of figure types as a constant to avoid using magic numbers. This will improve code readability and maintainability. For example, you can define `private static final int FIGURE_COUNT = 5;` and use it in `random.nextInt(FIGURE_COUNT)`.